### ISSUE TITLE
Can you give me a sample about the function HoughCircles?

### ISSUE THREAD

**Issue by boboyan** [hybridgroup/gocv#49]
```
gocv.CvtColor(img, edges, gocv.ColorBGRToGray)
gocv.GaussianBlur(edges, edges, image.Pt(9, 9), 2, 2, gocv.BorderDefault)
gocv.HoughCircles(edges, circles, 3, 1.5, float64(edges.Rows()/8))
fmt.Println(circles.Mean(), circles.Rows(), circles.Cols())
```

I want to detect circles.
I can't use the HoughCircles because no samples for HoughCircles.

How to use the `circles`?

please help me, tks

**Comment by deadprogram** [comment]
Hello @boboyan you might want to look here: https://github.com/hybridgroup/gocv/blob/master/imgproc_test.go#L269

Hope that helps!

**Comment by boboyan** [comment]
I know this, but my problem is how to use the circles, I want to get the every circle. 

For example this C code:
```
        vector<Vec3f> circles;
        //霍夫圆
        HoughCircles(edges, circles, CV_HOUGH_GRADIENT, 1.5, 10, 200, 100, 0, 0);
        for (size_t i = 0; i < circles.size(); i++)
        {
            Point center(cvRound(circles[i][0]), cvRound(circles[i][1]));
            int radius = cvRound(circles[i][2]);
            //绘制圆心  
            circle(frame, center, 3, Scalar(0, 255, 0), -1, 8, 0);
            //绘制圆轮廓  
            circle(frame, center, radius, Scalar(155, 50, 255), 3, 8, 0);
        }
```

**Comment by deadprogram** [comment]
This is a similar example although using HoughLinesP: https://github.com/hybridgroup/gocv/blob/master/imgproc_test.go#L350

**Comment by boboyan** [comment]
so sorry, what means for the col and row.

**Comment by deadprogram** [comment]
This previous issue might help get you started: https://github.com/hybridgroup/gocv/issues/35#issuecomment-349233299

**Comment by deadprogram** [comment]
Additional sample and WithParams variation have been released as part of v0.11.0 thank you everyone! Now closing.

**Comment by minskiter** [comment]
> I know this, but my problem is how to use the circles, I want to get the every circle.
> 
> For example this C code:
> 
> ```
>         vector<Vec3f> circles;
>         //霍夫圆
>         HoughCircles(edges, circles, CV_HOUGH_GRADIENT, 1.5, 10, 200, 100, 0, 0);
>         for (size_t i = 0; i < circles.size(); i++)
>         {
>             Point center(cvRound(circles[i][0]), cvRound(circles[i][1]));
>             int radius = cvRound(circles[i][2]);
>             //绘制圆心  
>             circle(frame, center, 3, Scalar(0, 255, 0), -1, 8, 0);
>             //绘制圆轮廓  
>             circle(frame, center, radius, Scalar(155, 50, 255), 3, 8, 0);
>         }
> ```

``` go
        circles := gocv.NewMat()
	defer circles.Close()
	gocv.HoughCirclesWithParams(origin, &circles, gocv.HoughGradient, 1, 60, 100, 30, 5, 30)
	for j := 0; j < circles.Cols(); j++ {
		vec := circles.GetVecfAt(0, j)
		gocv.Circle(&mat, image.Point{
			X: int(vec[0]),
			Y: int(vec[1]),
		}, int(vec[2]), color.RGBA{R: 255, G: 0, B: 0, A: 0}, 3)
	}
```
